analyzers/src/SonarAnalyzer.CSharp/Rules/DatabasePasswordsShouldBeSecure.cs

```diff
@@ -0,0 +1,116 @@
+        {
+            { "UseSqlServer", ImmutableArray.Create(KnownType.Microsoft_EntityFrameworkCore_DbContextOptionsBuilder, KnownType.Microsoft_EntityFrameworkCore_SqlServerDbContextOptionsExtensions) },
+            // the namespaces are different in .NET Core 3.1 and .NET 5
+            { "UseNpgsql", ImmutableArray.Create(KnownType.Microsoft_EntityFrameworkCore_DbContextOptionsBuilder, KnownType.Microsoft_EntityFrameworkCore_NpgsqlDbContextOptionsExtensions, KnownType.Microsoft_EntityFrameworkCore_NpgsqlDbContextOptionsBuilderExtensions) },
+            { "UseMySQL", ImmutableArray.Create(KnownType.Microsoft_EntityFrameworkCore_DbContextOptionsBuilder, KnownType.Microsoft_EntityFrameworkCore_MySQLDbContextOptionsExtensions)  },
+            { "UseSqlite", ImmutableArray.Create(KnownType.Microsoft_EntityFrameworkCore_DbContextOptionsBuilder, KnownType.Microsoft_EntityFrameworkCore_SqliteDbContextOptionsBuilderExtensions)  },
+            { "UseOracle", ImmutableArray.Create(KnownType.Microsoft_EntityFrameworkCore_DbContextOptionsBuilder, KnownType.Microsoft_EntityFrameworkCore_OracleDbContextOptionsExtensions)  },
+        };
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        protected override void Initialize(SonarAnalysisContext context) =>
+            context.RegisterSyntaxNodeActionInNonGenerated(c =>
+                {
+                    var invocation = (InvocationExpressionSyntax)c.Node;
+                    if (IsConnectionStringInvocation(invocation, c.SemanticModel)
+                        && HasEmptyPasswordArgument(invocation.ArgumentList.Arguments))
+                    {
+                        c.ReportDiagnosticWhenActive(Diagnostic.Create(Rule, invocation.GetLocation()));
+                    }
+                },
+                SyntaxKind.InvocationExpression);
+
+        private static bool IsConnectionStringInvocation(InvocationExpressionSyntax invocation, SemanticModel semanticModel) =>
+            invocation.Expression is MemberAccessExpressionSyntax memberAccess
+            && memberAccess.GetName() is { } name
+            && NamesWithTypes.ContainsKey(name)
+            && IsInvocationOn(memberAccess, semanticModel, NamesWithTypes[name]);
+
+        private static bool IsInvocationOn(MemberAccessExpressionSyntax memberAccess, SemanticModel semanticModel, ImmutableArray<KnownType> knownTypes) =>
+            semanticModel.GetSymbolInfo(memberAccess).Symbol is { } symbol
+            && symbol.ContainingType.DerivesFromAny(knownTypes);
+
+        private static bool HasEmptyPasswordArgument(SeparatedSyntaxList<ArgumentSyntax> argumentList) =>
+            ConnectionStringArgument(argumentList)?.Expression switch
+            {
+                LiteralExpressionSyntax literal when literal.IsKind(SyntaxKind.StringLiteralExpression) => IsVulnerable(literal.Token.ValueText) && !HasSanitizers(literal.Token.ValueText),
```

Comment:
Do we need this `when literal.IsKind(SyntaxKind.StringLiteralExpression)`? If I remove it and run the tests they remain green.